scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxyActiveObject.java

```diff
@@ -137,8 +138,13 @@ public void releaseNodes(NodeSet nodes, Script<?> cleaningScript, Map<String, Se
             if (cleaningScript == null) {
                 releaseNodes(nodes);
             } else {
-                for (Node node : nodes) {
-                    handleCleaningScript(node, cleaningScript, variables,genericInformation,taskId);
+                if (InternalTask.isScriptAuthorized(taskId, cleaningScript)) {
+                    for (Node node : nodes) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)